pyquil/wavefunction.py

```diff
@@ -1,27 +1,96 @@
 """
 Module containing the Wavefunction object and methods for working with wavefunctions.
 """
+import struct
+import warnings
+
+import numpy as np
+from six import integer_types
+
+OCTETS_PER_DOUBLE_FLOAT = 8
+OCTETS_PER_COMPLEX_DOUBLE = 2 * OCTETS_PER_DOUBLE_FLOAT
 
 
 class Wavefunction(object):
 
-    def __init__(self, amplitude_vector):
+    def __init__(self, amplitude_vector, classical_memory=None):
         """
         Initializes a wavefunction
         :param amplitude_vector: A numpy array of complex amplitudes
+        :param classical_memory: Any classical memory associated with this wavefunction result
         """
         if len(amplitude_vector) == 0 or len(amplitude_vector) & (len(amplitude_vector) - 1) != 0:
             raise TypeError("Amplitude vector must have a length that is a power of two")
 
         self.amplitudes = amplitude_vector
+        self.classical_memory = classical_memory
+
+    @staticmethod
+    def ground(qubit_num):
```

Comment:
"ground" and "groundstate" are physics inspired terms that we should refrain from using (including "excited state").  Better would be to call this `zeros` (like after the numpy function) or `all_zeros`.  This nomenclature is more self-explanatory.